### ISSUE TITLE
Add an option to output all the nodes with 'tree-sitter parse'

### ISSUE THREAD

**Issue by mjambon** [tree-sitter/tree-sitter#1154]
My ideal workflow for inspecting the output of a parser would be:
```
<edit grammar.js>
<edit example>
$ tree-sitter generate
$ tree-sitter parse example
```
For this, I need to see all the nodes of the tree as provided by the C API. Here's what we get right now:
```
$ cat grammar.js 
module.exports = grammar({
  name: 'debug',
  rules: {
    program: $ => choice(
      'a',  // ok
      /b/,  // missing node
    ),
  }
});
$ echo a > a
$ tree-sitter generate
$ tree-sitter parse a
(program [0, 0] - [1, 0])
$ tree-sitter parse a -x
(program [0, 0] - [1, 0])
<program>a</program>

```
Neither the default output nor the XML output show the node that matches `'a'` even though it exists.

Here's what `tree-sitter parse` currently offers:
```
$ tree-sitter parse --help
tree-sitter-parse 
Parse files

USAGE:
    tree-sitter parse [FLAGS] [OPTIONS] [--] [paths]...

FLAGS:
    -d, --debug          
    -D, --debug-graph    
    -x, --xml            
    -h, --help           Prints help information
    -q, --quiet          
    -s, --stat           
    -t, --time           
    -V, --version        Prints version information

OPTIONS:
    -e, --edit <edits>...       
        --paths <paths-file>    
        --scope <scope>         
        --timeout <timeout>     

ARGS:
    <paths>...    
```
The default output only shows names for nodes corresponding to named rules. I work on [ocaml-tree-sitter](https://github.com/returntocorp/ocaml-tree-sitter), which takes the CST and recovers a typed CST. This requires inspecting the full structure of the tree, including all the nodes corresponding to tokens. The way we've been doing this is by having our generated ocaml code dump a json view of the tree. It would be much more convenient for us, and presumably to authors of other projects to be able to dump such detailed view of the tree with `tree-sitter parse`, without writing extra code and preferably without setting up a nodejs project.

Such output would be quite verbose, so it would be good to have it as an option and not as the default. We could extend the XML output but XML is awkward to work with and would get a json tree. I suggest adding a `--json`/`-j` option that would output all the data available from the C API.

**Comment by ahlinc** [comment]
I'm in process implementing of new tree rendering for tree sitter that will include many representations, current rendering of the tree you can see in a picture in this PR https://github.com/ahlinc/tree-sitter/pull/4.
@mjambon you may tr to use rendering from the [new-tree-rendering](https://github.com/ahlinc/tree-sitter/tree/new-tree-rendering) branch. I'd be appreciate for any feedback.

**Comment by WillLillis** [comment]
Implemented in #3762 

![image](https://github.com/user-attachments/assets/6cb7e30c-93df-4e36-888f-93248d914db8)
